### DISCUSSION TITLE
Open modal info after create Action

### DISCUSSION THREAD

**Discussion by evaldorcardoso** [filamentphp/filament#13482] [Help] [answered]
### Package

Actions

### Package Version

v3.2

### How can we help you?

I would like to open a modal with some info to be called inside the hook "after" attached to the CreateAction, this occurs after the record is saved to database.
I'm using "simple" resource to manage my records.

**Comment by leandrocfe** [comment]
Maybe this?
https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component#chaining-actions

**Reply by evaldorcardoso** [reply]
@leandrocfe  Thanks for repling.

I've tried the chainning-actions with "replaceMountedAction" but it replaces the original action and the $record is not saved to database. I would like to keep the 2 actions, first save to database and after call another action to open a modal.

**Reply by leandrocfe** [reply, marked as answer]
```php
Actions\CreateAction::make()
    ->registerModalActions([
        Actions\Action::make('anotherAction')
            ->form([
                //...
            ])
            ->action(function () {
                //do something...
            })
            ->cancelParentActions(),
    ])
    ->after(fn () => $this->mountAction('anotherAction'))
```

**Reply by evaldorcardoso** [reply]
Thanks, that works, the modal opens.
@leandrocfe just one more question, there is a way to pass the arguments from the $record after created to the "anotherAction" action and show them in the modal?

My use case is tho create api keys and just after create show a modal with the generated client_secret just one time.
Thanks in advance.

**Reply by leandrocfe** [reply]
try to inject `array $arguments`
https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component#chaining-actions

**Reply by evaldorcardoso** [reply]
Thanks for answering, worked like a charm!

**Reply by nilskretschmer** [reply]
Is it possible that this does not work with table actions like in a RelationManager`s `headerActions`?

**Comment by xiaomlove** [comment]
Does anyone have more details on this? I need to receive the return value of the action in after to make a modal display and close the modal of the original action.

**Comment by plmrlnsnts** [comment]
In my case, I want to open the modal on page load. This can be achieved using the snippet below:

```
->extraAttributes(fn (Action $action) => [
    'wire:init' => $action->getLivewireClickHandler(),
])
```
